### DISCUSSION TITLE
Assessment of ASME BPVC III Div. 5 (2023 Ed.) HBB-Z appendix, 316 SS constitutive law within MFront and Code_Aster

### DISCUSSION THREAD

**Discussion by Irvise** [thelfer/tfel#606] [General]
Hi Thomas,

as I mentioned in an issue, I wanted to see if MFront would be a nice, easy system on which to build the behavioural law for 316 SS as defined in Div. 5 (High Temperature reactors). I focused on the built-in bricks as that is the first stop that I wanted to analyse. I also took a look at what is already available in Code_Aster just to have a wider analysis area.

Since the information is quite large, I made a report. Its quality is lacking as I have done very little proof reading and typos/mistakes are sure to be found. Nonetheless, I think it is a good starting point. You can skip the text and go directly to the "Conclusions" sections directly if you wish.

[report.pdf](https://github.com/user-attachments/files/16479327/report.pdf)

I am sharing this with you and the community in case you find it interesting, use it as a reference or would like to discuss it further.

Best regards,
Fer

**Comment by Irvise** [comment]
Ouch, I had not updated the TOCs nor the acronyms... Here is an updated version 
[report.pdf](https://github.com/user-attachments/files/16479341/report.pdf)

**Comment by thelfer** [comment]
Hi @Irvise,

I think that you can take example of this tutorial: https://thelfer.github.io/tfel/web/isotropicplasticityamstrongfrederickinematichardening.html

There are two main changes:

1. taking damage into account
2. the evolution of the radius of the yields surface

You have to introduce two new state variables. Since variables and residuals must be scaled by hand, I recommand that you choose `d` (for the damage) and `rR = sigma1/young`.

The computation of the consistent tangent operator must be changed quite a bit as explained here: https://thelfer.github.io/tfel/web/implicit-dsl.html#computation-of-the-consistent-tangent-operator
You can take inspiration from the Hayurst example: https://github.com/thelfer/tfel/blob/master/mfront/tests/behaviours/ImplicitHayhurst.mfront
